lamindb/_query_set.py

```diff
@@ -604,17 +605,64 @@ def first(self) -> Record | None:
             return None
         return self[0]
 
+    def _handle_unknown_field(self, error: FieldError) -> None:
+        """Suggest available fields if an unknown field was passed."""
+        if "Cannot resolve keyword" in str(error):
+            field = str(error).split("'")[1]
+            fields = ", ".join(
+                sorted(
+                    f.name
+                    for f in self.model._meta.get_fields()
+                    if not f.name.startswith("_")
+                    and not f.name.startswith("links_")
+                    and not f.name.endswith("_id")
+                )
+            )
+            raise FieldError(
+                f"Unknown field '{field}'. Available fields: {fields}"
+            ) from None
+        raise error  # pragma: no cover
+
     def get(self, idlike: int | str | None = None, **expressions) -> Record:
         """Query a single record. Raises error if there are more or none."""
-        return get(self, idlike, **expressions)
+        try:
+            return get(self, idlike, **expressions)
+        except ValueError as e:
+            # Pass through original error for explicit id lookups
+            if "Field 'id' expected a number" in str(e):
```

Comment:
This looks brittle but I guess you're right that Django doesn't have error types with this granularity.